Assets/MixedRealityToolkit.Tests/PlayModeTests/InteractableTests.cs

```diff
@@ -37,11 +39,470 @@ public IEnumerator TestAddInteractableAtRuntime()
+        /// Assembles a push button from primitives and uses simulated hand input to press it.
+        /// </summary>
+        /// <returns></returns>
+        //[UnityTest] Temporarily ignoring this test until we can get animation working
+        public IEnumerator TestSimulatedHandInputOnRuntimeAssembled()
+        {
+            // instantiate scene
+            TestUtilities.InitializeMixedRealityToolkitAndCreateScenes(true);
+            TestUtilities.InitializePlayspace();
+
+            // Load interactable prefab
+            GameObject interactableObject;
+            Interactable interactable;
+            Transform translateTargetObject;
+
+            AssembleInteractableButton(
+                out interactableObject,
+                out interactable,
+                out translateTargetObject);
+
+            Vector3 targetStartPosition = translateTargetObject.transform.localPosition;
+
+            yield return null;
+
+            // Add a touchable and configure for touch events
+            NearInteractionTouchable touchable = interactableObject.AddComponent<NearInteractionTouchable>();
+            touchable.EventsToReceive = TouchableEventType.Touch;
+            touchable.Bounds = Vector2.one;
+            touchable.SetLocalForward(Vector3.up);
+            touchable.SetLocalUp(Vector3.forward);
+            touchable.LocalCenter = Vector3.up * 2.75f;
+
+            // Add a touch handler and link touch started / touch completed events
+            TouchHandler touchHandler = interactableObject.AddComponent<TouchHandler>();
+            touchHandler.OnTouchStarted.AddListener((HandTrackingInputEventData e) => interactable.SetInputDown());
+            touchHandler.OnTouchCompleted.AddListener((HandTrackingInputEventData e) => interactable.SetInputDown());
+
+            touchHandler.OnTouchStarted.AddListener((HandTrackingInputEventData e) => Debug.Log("SetInputDown"));
+            touchHandler.OnTouchCompleted.AddListener((HandTrackingInputEventData e) => Debug.Log("SetInputUp"));
+
+            // Move the camera to origin looking at +z to more easily see the button.
+            MixedRealityPlayspace.PerformTransformation(
+            p =>
+            {
+                p.position = Vector3.zero;
+                p.LookAt(Vector3.forward);
+            });
+
+            // Move the hand forward to intersect the interactable
+            var inputSimulationService = PlayModeTestUtilities.GetInputSimulationService();
+            int numSteps = 32;
+            Vector3 p1 = new Vector3(0.0f, 0f, 0f);
+            Vector3 p2 = new Vector3(0.05f, 0f, 0.51f);
+            Vector3 p3 = new Vector3(0.0f, 0f, 0.0f);
+
+            yield return PlayModeTestUtilities.ShowHand(Handedness.Right, inputSimulationService);
+            yield return PlayModeTestUtilities.MoveHandFromTo(p1, p2, numSteps, ArticulatedHandPose.GestureId.Poke, Handedness.Right, inputSimulationService);
+            yield return new WaitForSeconds(1f);
```

Comment:
Why do we need to wait a second here?

Reply:
Mainly to make the test easier to eyeball in the editor, but also to give the interactable theme time to translate its target. I can cut it down if we want to keep the tests brief.

Reply:
Instead of looking at transform movement, can we just add a handler to OnClick method?

Reply:
when validating the tests you can use `PlayModeTestUtilities.WaitForEnterKey` to actually wait until a key is pressed. I'd prefer to keep the test as short as possible (max 0.5 seconds) since if every test takes a second they will add up.